client-v2/src/shared/components/snapLink/SnapLink.tsx

```diff
@@ -64,6 +74,7 @@ const SnapLink = ({
     (articleFragment.meta.customKicker
       ? `{ ${articleFragment.meta.customKicker} }`
       : 'No headline');
+  const urlPath = articleFragment.meta.href;
```

Comment:
For some diabolical reason we store Guardian paths without the guardian domain. Presumably so if the domain changes the links are OK. (`30X`?) For this reason, you'll need to add the appropriate domain if there isn't one, and we can crib from v1 here -- 

```
const normaliseSnapUrl = (href: string) => {
  if (href && !/^https?:\/\//.test(href)) {
    return 'https://' + urls.guDomain + href;
  }

  return href;
};
```

where urls.guDomain can be set in e.g. `constants/urls`.

(If this sounds oddly specific, I touched this task this afternoon, but you were faster 🚀 👍)

Reply:
To test this, drag e.g. https://www.theguardian.com/uk into the tool. You should see a lonely `/uk` pop up in the snap, that links, inappropriately, to the tool itself.

Reply:
Aha. Is that regex just validating whether we have a complete URL or a partial path, as would be the case with Guardian links?

Reply:
precisely -- it's logic that exists in v1 but it's obviously encoding a big assumption about our model (incomplete paths belong to the guardian!)